A localized Windows user directories with spaces prevents starting the dev environment

### Description

I'm using a lithuanian version of Windows 11. My home directory is "Saulius S24" (Windows is connected to mobile device). 
I'm following the getting started guide for WP dev env setup and I'm stuck with the script throwing an error in PowerShell terminal:
```
PS C:\Users\Toshiba\CrossDevice\Saulius S24\code\gutenberg> npm run dev

> gutenberg@23.3.1 dev
> npm run --workspace @wordpress/build-scripts dev --


> @wordpress/build-scripts@0.0.0 dev
> node ./dev.mjs

'ps' is not recognized as an internal or external command,
operable program or batch file.
🔨 Starting development build...

🔍 Checking dependencies...
(node:6772) [DEP0190] DeprecationWarning: Passing args to a child process with shell option true can lead to security vulnerabilities, as the arguments are not escaped, only concatenated.
(Use `node --trace-deprecation ...` to show where the warning was created)

   ✔ All good.

🧹 Cleaning packages...

📦 Building workspaces...
node:internal/modules/cjs/loader:1503
  throw err;
  ^

Error: Cannot find module 'C:\Users\Toshiba\CrossDevice\Saulius'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1500:15)
    at wrapResolveFilename (node:internal/modules/cjs/loader:1071:27)
    at defaultResolveImplForCJSLoading (node:internal/modules/cjs/loader:1095:10)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1116:12)
    at Module._load (node:internal/modules/cjs/loader:1285:25)
    at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
    at node:internal/main/run_main_module:33:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v24.16.0

❌ Dev build failed: Command failed: node C:\Users\Toshiba\CrossDevice\Saulius S24\code\gutenberg\tools\build-scripts\packages\generate-worker-placeholders.mjs
npm error Lifecycle script `dev` failed with error:
npm error code 1
npm error path C:\Users\Toshiba\CrossDevice\Saulius S24\code\gutenberg\tools\build-scripts
npm error workspace @wordpress/build-scripts@0.0.0
npm error location C:\Users\Toshiba\CrossDevice\Saulius S24\code\gutenberg\tools\build-scripts
npm error command failed
npm error command C:\WINDOWS\system32\cmd.exe /d /s /c node ./dev.mjs
```

So what happens is due to the home directory "Saulius S24" having a space it is being truncated or split using the space and the further script is having a problem with directory path resolution.

### Step-by-step reproduction instructions

Have a home directory with spaces.

Follow the instructions of https://developer.wordpress.org/block-editor/contributors/code/getting-started-with-code-contribution/

run `npm install`, `npm run dev` in the PowerShell.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

Edition	Windows 11 Pro
Version	25H2
OS version	26200.7922
1000.26100.300.0

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

--- a/tools/build-scripts/dev.mjs
+++ b/tools/build-scripts/dev.mjs
@@ -3,7 +3,8 @@
 /**
  * External dependencies
  */
-import { execSync, spawn } from 'child_process';
+import { execSync } from 'child_process';
+import spawn from 'cross-spawn';
 import { fileURLToPath } from 'url';
 import path from 'path';
 import fs from 'fs';
@@ -28,7 +29,6 @@ function exec( command, args = [], options = {} ) {
 		const childOptions = {
 			cwd: ROOT_DIR,
 			stdio: silent ? 'pipe' : 'inherit',
-			shell: true,
 			...spawnOptions,
 		};
 
@@ -89,7 +89,6 @@ function execAsync( command, args = [], options = {} ) {
 	return spawn( command, args, {
 		cwd: ROOT_DIR,
 		stdio: 'inherit',
-		shell: true,
 		...options,
 	} );
 }
@@ -146,7 +145,7 @@ async function dev() {
 
 		// This must happen before TypeScript compilation because some packages
 		// (like vips) have source files that import from generated worker-code.ts
-		await exec( 'node', [
+		await exec( process.execPath, [
 			path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ),
 		] );
 
@@ -162,7 +161,7 @@ async function dev() {
 		console.log( `   ✔ Built TypeScript types (${ buildTime }ms)` );
 
 		console.log( '\n✅ Checking type declaration files...' );
-		await exec( 'node', [
+		await exec( process.execPath, [
 			path.join(
 				__dirname,
 				'packages/check-build-type-declaration-files.cjs'
@@ -170,7 +169,7 @@ async function dev() {
 		] );
 
 		console.log( '\n📦 Building vendor files...' );
-		await exec( 'node', [
+		await exec( process.execPath, [
 			path.join( __dirname, 'packages/build-vendors.mjs' ),
 		] );
 
@@ -198,7 +197,6 @@ async function dev() {
 		const buildWatch = spawn( 'wp-build', [ '--watch' ], {
 			cwd: ROOT_DIR,
 			stdio: [ 'inherit', 'pipe', 'inherit' ],
-			shell: true,
 			env: { ...process.env, NODE_ENV: 'development' },
 		} );
 

--- a/tools/build-scripts/dev.mjs
+++ b/tools/build-scripts/dev.mjs
@@ -145,7 +145,7 @@ async function dev() {
 
 		// This must happen before TypeScript compilation because some packages
 		// (like vips) have source files that import from generated worker-code.ts
-		await exec( process.execPath, [
+		await exec( 'node', [
 			path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ),
 		] );
 
@@ -161,7 +161,7 @@ async function dev() {
 		console.log( `   ✔ Built TypeScript types (${ buildTime }ms)` );
 
 		console.log( '\n✅ Checking type declaration files...' );
-		await exec( process.execPath, [
+		await exec( 'node', [
 			path.join(
 				__dirname,
 				'packages/check-build-type-declaration-files.cjs'
@@ -169,7 +169,7 @@ async function dev() {
 		] );
 
 		console.log( '\n📦 Building vendor files...' );
-		await exec( process.execPath, [
+		await exec( 'node', [
 			path.join( __dirname, 'packages/build-vendors.mjs' ),
 		] );
 

